designdocs/recovery_system.md

```diff
@@ -0,0 +1,587 @@
+
+4. New Control Key Registration
+
+    The Recovery Provider registers a new Control Key for the MSA. There are two approaches:
+
+    - User-specified key: If the user is somewhat technical, they might generate a new key pair on their own device (e.g. via a wallet app) and only give the public key to the Recovery Provider. This is ideal for security, as the private key never leaves the user. The Recovery Provider would ask “Please paste a new public key to add to your account.” However, this requires the user to know how to create a key pair, which may not be user-friendly in an emergency.
+    - Provider-generated key: The Recovery Provider can create a new key pair on behalf of the user. For instance, it generates a new Ed25519, Sr25519, or Ethereum key (compatible with Frequency). The Recovery Provider may then securely transmit the private key, or Provider Approved Login Credentials to the user after the on-chain update. This approach is simpler for the user but requires trust that the Recovery Provider deletes its copy. In practice the Recovery Provider could generate a mnemonic and show it to the user over a secure TLS connection for them to copy, minimizing its own storage of the key.
+
+    This design recommends giving the user the option, but in either case, a public key for the new control key is decided at this step.
+
+5. On-Chain Recovery Execution
+
+    The Recovery Provider now submits a special Recovery extrinsic to the Frequency blockchain. This transaction does two main things atomically:
+
+    - Invalidate Old Recovery Hash: It clears the hash so it cannot be reused.
+    - Add New Control Key: It adds the new Control Key to the MsaId.
+
+6. Completion and User Notification
+
+    The blockchain emits an event confirming the Control Key addition. The Recovery Provider sees the transaction succeed and now communicates the results to the user. It may provide the user with:
+
+    - Their MSA ID, if the user didn’t know it (users may not normally need to know their numeric MSA ID, but for transparency the Recovery Provider should tell them which account was recovered).
+    - The new Control Key credentials or Provider Approved Login Credentials.
+
+    At this point, the user can use a wallet, possibly the recovery provider’s interface or any Frequency-compatible wallet, to load the new Control Key and access their account (MSA). They should see that their data, messages, and any delegations under that MSA are intact. They have essentially regained control without needing the original provider.
+
+    Additionally, the governance or Frequency network can record that “Recovery Provider X performed recovery for MSA Y at time Z.” Recovery Providers are trusted, but such events may be monitored to ensure no Recovery Provider is abusing the system. If a Recovery Provider is found adding keys without proper user requests, governance can revoke their recovery role. This provides an accountability layer.
```

Comment:
If we end up asking for a bonded stake to be a recovery provider then the penalty needs to include slashing.  If they're a Provider, governance might consider revoking their Provider status, slashing their stake for Capacity, or otherwise disallowing new delegations.